### PR TITLE
Create getbit.py

### PR DESCRIPTION
**Pull request by rag9704**
Numpy equivalent of Matlab's getbit(n,x) to return the bit value at position x of n (in decimal)

### CODE REVIEW

**Comment by eric-wieser** [comment]
Thanks for the contribution.

Typically, new functions added to numpy should first go through the mailing list [numpy-discussion@python.org](https://mail.python.org/mailman/listinfo/numpy-discussion). We're realizing that our contribution guidelines are lacking in explaining this (#13924), so apologies if this has resulted in the wasting of your time.

---

My feeling would be you will get little support for this function, primarily because you can write it without using numpy at all:
```python
def bitget(value, bit_no):
    # note: use `!= 0` instead of `bool` to make this work on numpy arrays
    return (b & (1 << bit_no)) != 0
```

However, you might find that there is sufficient interest on either our mailing list or a matlab mailing list for a `matlab_functions` python-package, that uses matlab names for numpy / pure-python things

File: numpy/matlabequivalent.py [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import numpy as np
+
+#contribution no.1      08/07/2019
+#matlab equivalent of bitget
+def bitget(value,bit_no):                  # equivalent bitget in matlab!!
+    binval=eval(np.binary_repr(value))de
```
Comment (eric-wieser) [inline numpy/matlabequivalent.py]: Typo here.

Also, as a general rule, the use of `eval` should raise red flags - even if it's not a security issue (in this case, it's safe), its often a symptom of there being a better way to solve the problem

Comment (rag9704) [reply numpy/matlabequivalent.py]: Sorry for the typo! but, there is no such function getbit in Numpy/python as there is in Matlab! 
Should I send it to the mailing list? I am new to the open-source!

**Comment by eric-wieser** [comment]
> there is no such function getbit in Numpy/python as there is in Matlab!

I think there's a more general question to be asked here - is it goal of numpy to provide every function that matlab provides (in order to ease users transitioning between the languages)?

My leaning is towards saying that is a non-goal, but whatever the decision is it ought to be written down.

Asking the mailing list for general opinions on when matlab functions should be added within numpy, and then giving `getbit` as an example might be a good place to start.

**Comment by rag9704** [comment]
Thanks man!
Much appreciated



[image: Mailtrack]
<https://mailtrack.io?utm_source=gmail&utm_medium=signature&utm_campaign=signaturevirality5&>
Sender
notified by
Mailtrack
<https://mailtrack.io?utm_source=gmail&utm_medium=signature&utm_campaign=signaturevirality5&>
07/08/19,
02:31:28 PM

**Comment by bashtage** [comment]
This function only exists in MATLAB because MATLAB doesn't support shirt notation with << or >>, and so uses a function instead.  Much less essential in Python.

**Pull request closed without merging**